src/plugins/intel_cpu/src/async_infer_request.cpp

```diff
@@ -9,9 +9,24 @@
 ov::intel_cpu::AsyncInferRequest::AsyncInferRequest(
     const std::shared_ptr<IInferRequest>& request,
     const std::shared_ptr<ov::threading::ITaskExecutor>& task_executor,
-    const std::shared_ptr<ov::threading::ITaskExecutor>& callback_executor)
-    : ov::IAsyncInferRequest(request, task_executor, callback_executor) {
+    const std::shared_ptr<ov::threading::ITaskExecutor>& callback_executor,
+    const bool is_single_thread)
+    : ov::IAsyncInferRequest(request, task_executor, callback_executor),
+      m_internal_request(request),
+      m_task_executor(task_executor) {
     static_cast<SyncInferRequest*>(request.get())->set_async_request(this);
+    m_infer_func = [this]() {
+        ov::IAsyncInferRequest::infer();
+    };
+    if (is_single_thread) {
+        m_infer_func = [this]() {
+            check_tensors();
+            auto streams_executor = std::dynamic_pointer_cast<ov::threading::IStreamsExecutor>(m_task_executor);
```

Comment:
It's not a good idea to perform the dynamic cast every time.
It's better to store the specific type:
`std::shared_ptr<ov::threading::IStreamsExecutor> m_stream_executor;` istead of the `m_task_executor` (which is not used directly).

Reply:
updated